aries_vcx/src/new_protocols/mod.rs

```diff
@@ -0,0 +1,355 @@
+#![allow(unused)]
+
+use aries_vcx_core::{
+    ledger::base_ledger::{AnoncredsLedgerRead, IndyLedgerRead},
+    wallet::base_wallet::BaseWallet,
+};
+
+use diddoc_legacy::aries::diddoc::AriesDidDoc;
+use messages::msg_fields::protocols::connection::{
+    invitation::{InvitationContent, PairwiseDidInvitationContent, PairwiseInvitationContent, PublicInvitationContent},
+    request::RequestContent,
+    response::ResponseContent,
+    ConnectionData,
+};
+use url::Url;
+
+use crate::{
+    common::{
+        ledger::transactions::get_service,
+        signing::{decode_signed_connection_response, sign_connection_response},
+    },
+    errors::error::VcxResult,
+};
+
+pub struct BootstrapInfo {
+    service_endpoint: Url,
+    recipient_keys: Vec<String>,
+    routing_keys: Vec<String>,
+    did: Option<String>,
+    service_endpoint_did: Option<String>,
+}
+
+async fn parse_public_invitation<T>(ledger_read: &T, invitation: PublicInvitationContent) -> VcxResult<BootstrapInfo>
+where
+    T: IndyLedgerRead + AnoncredsLedgerRead,
+{
+    let service = match get_service(ledger_read, &invitation.did).await {
+        Ok(s) => s,
+        Err(err) => {
+            error!("Failed to obtain service definition from the ledger: {}", err);
+            return Err(err);
+        }
+    };
+
+    let info = BootstrapInfo {
+        service_endpoint: service.service_endpoint,
+        recipient_keys: service.recipient_keys,
+        routing_keys: service.routing_keys,
+        did: Some(invitation.did),
+        service_endpoint_did: None,
+    };
+
+    Ok(info)
+}
+
+fn parse_pw_invitation(invitation: PairwiseInvitationContent) -> BootstrapInfo {
+    BootstrapInfo {
+        service_endpoint: invitation.service_endpoint,
+        recipient_keys: invitation.recipient_keys,
+        routing_keys: invitation.routing_keys,
+        did: None,
+        service_endpoint_did: None,
+    }
+}
+
+async fn parse_pw_did_invitation<T>(
+    ledger_read: &T,
+    mut invitation: PairwiseDidInvitationContent,
+) -> VcxResult<BootstrapInfo>
+where
+    T: IndyLedgerRead + AnoncredsLedgerRead,
+{
+    let service = match get_service(ledger_read, &invitation.service_endpoint).await {
```

Comment:
just noticed - it should be more like 
```
&invitation.content.services[0]
```
instead of 
```
&invitation.service_endpoint
``
(correct, although not ideal)